engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -247,17 +356,18 @@ public function get($guid, $type = '') {
 	 * has changed.
 	 *
 	 * @param int $guid The GUID of the entity
-	 *
 	 * @return bool
 	 */
 	public function exists($guid) {
 
-		$query = "SELECT 1 FROM {$this->db->prefix}entities
-			WHERE guid = :guid";
-		
-		$result = $this->db->getDataRow($query, false, [
-			':guid' => (int) $guid,
-		]);
+		// need to ignore access and show hidden entities to check existence
+		$ia = $this->session->setIgnoreAccess(true);
+		$show_hidden = access_show_hidden_entities(true);
+
+		$result = $this->getRow($guid);
```

Comment:
The only problem with this is someone adding `and` clauses to the access sql. Should we even be worrying about someone ignoring ignored access in the hook handler?